llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -9934,6 +9934,12 @@ SDValue SITargetLowering::LowerINTRINSIC_VOID(SDValue Op,
     auto NewMI = DAG.getMachineNode(Opc, DL, Op->getVTList(), Ops);
     return SDValue(NewMI, 0);
   }
+  case Intrinsic::amdgcn_s_prefetch_data: {
+    // For non-global address space preserve the chain and remove the call.
+    if (!AMDGPU::isFlatGlobalAddrSpace(cast<MemSDNode>(Op)->getAddressSpace()))
+      return Op.getOperand(0);
+    return Op;
```

Comment:
I'd expect the private/local cases to be an error. Also collectFlatAddressOperands should handle this, to get the flat->global optimization in InferAddressSpaces

Reply:
Are you sure you want this? Prefetch is a hint, I did not want a hard error, just remove the nonsense. That is because after a set of inline and infer address space the address may turn to be non-global, but shall it fail to compile?

Reply:
Besides it is not this change.

Reply:
Infer can just not do the change if the resulting address space isn't global

Reply:
For the context of this change what do I have to do? This is the code the change does not touch.

Reply:
I understand that the change is very similar to the previous one, but the keyword is buffer. It has nothing to do with infer.